### ISSUE TITLE
MultipleNullnessAnnotations shouldn't consider Validation Annotations

### ISSUE THREAD

**Issue by xenoterracide** [google/error-prone#4334]
The Jakarta (formerly JavaX) Validation spec shouldn't be considered for this bug. My rationale is that it is a runtime validation which may not occur until late in a process. 

This is not, perhaps the best example of code, but it's what my brain is currently coming up with.

```java
public class Foo {

  @NotNull
  @Nullable
  Long getId();
} 
```

```
if ( foo.getId() == null ) {
   foo.set(1L); // yes this is terrible, but you could query a database sequence, or this could be a UUID
}
repository.save(foo); // if I remove the above if statement, and a null is passed this would throw an error
```

so, getId is validly nullable at runtime, but at the point of validation we want it to be non null.

note: you could ignore the javax/jakarta namespaced annotations explicitly, but as far as I'm aware `NotNull` is always the name of the runtime checks, and some case insensitive variant of `NonNull` is the "compile time" variant

**Comment by cushon** [comment]
cc @cpovirk 

Thanks for the report. The use-case for MultipleNullnessAnnotations is primarily nullness annotations that have semantics similar to the https://github.com/jspecify/jspecify ones. We could look at having the check skip annotations with incompatible semantics, or clarifying what the use-case for it is.

**Comment by cpovirk** [comment]
Yes, thanks. I'm largely clueless about the validation annotations, which I have probably seen as many times in the past couple weeks as I had in my life previously.

We have at least one [list of nullness annotations](https://github.com/jspecify/jspecify-reference-checker/blob/75aa1de2a8f88f8c23347d86465844df9887874c/src/main/java/com/google/jspecify/nullness/NullSpecAnnotatedTypeFactory.java#L1995), which sadly includes at least one notably "`@NotNull`" annotation that is "similar to JSpecify"  more than "similar to Jakarta Validation," `org.jetbrains.annotations.NotNull`. That doesn't make this proposal impossible to implement, but it keeps it from being a one-liner. For now, you might want to set `-Xep:MultipleNullnessAnnotations:OFF`.

**Comment by xenoterracide** [comment]
Ugh, thanks jetbrains. Fortunately you can probably just use the two official validation spec paths for that. I don't know if I would bother going and tracking down other vendor specific ones. Like I'm sure hibernate had one at some point.
